src/modules/application/controllers/controllerEdit.js

```diff
@@ -0,0 +1,68 @@
+import { verifyToken } from "../../../utilities/jsonWebToken.js";
+import { databasePrisma } from "../../../prismaClient.js";
+
+export const handleEdit = async function (req) {
+  const { id: applicationId } = req.params;
+  const { applicant, listing, coverLetter, company } = req.body;
+
+  const token = req.headers.authorization;
+
+  let readyToken = token;
+  if (!token) {
+    return Promise.resolve({
+      status: 401,
+      message: "User has to be authenticated to make this request",
+    });
+  } else if (token.includes("Bearer")) {
+    readyToken = token.slice(7);
+  }
+
+  let verified;
+
+  if (readyToken !== undefined) {
+    try {
+      verified = verifyToken(readyToken);
+    } catch (error) {
+      return Promise.resolve({ status: 401, message: "Auth token not valid." });
+    }
+  }
```

Comment:
Currently we are allowing anyone logged in to change anyone else's applications, as well as change the listing, applicant and company that it was originally assigned to.

The token validation and user identity verification, for PUT should be the same as for DELETE, if you create a middleware function similar to how I have done for the listings you will be able to check and reuse it for both. You can look at the middleware I made here, and how I used it on the listing route.js.

https://github.com/NoroffFEU/agency-api.noroff.dev/tree/Company-Listings-Fixes/src/modules/listing/middleware

Reply:
Created middleware